src/SessionLocker.vala

```diff
@@ -62,38 +62,54 @@ public class Gala.SessionLocker : Clutter.Actor {
     private const string LOCK_PROHIBITED_KEY = "disable-lock-screen";
     private const string LOCK_ON_SUSPEND_KEY = "lock-on-suspend";
 
-    public signal void active_changed ();
+    public ScreenShield screen_shield { private get; construct; }
+
+    public int64 activation_time { get; private set; default = 0; }
 
     // Screensaver active but not necessarily locked
-    public bool active { get; private set; default = false; }
+    private bool _active = false;
+    public bool active {
+        get {
+            return _active;
+        }
+        private set {
+            if (!connected_to_buses) {
+                return;
+            }
 
-    public bool is_locked { get; private set; default = false; }
-    public bool in_greeter { get; private set; default = false; }
-    public int64 activation_time { get; private set; default = 0; }
+            if (_active != value) {
+                _active = value;
+                notify_property ("active");
```

Comment:
IIRC you don't actually have to notify manually (we don't do explicit notify https://docs.gtk.org/gobject/signal.Object.notify.html)